benchmark/src/lib.rs

```diff
@@ -298,8 +323,14 @@ impl Benchmarker {
         &mut self,
         txn_reqs: &[SubmitTransactionRequest],
         senders: &mut [AccountData],
+        pattern: Option<ConstantRate>,
     ) -> (usize, usize, u128, u128) {
-        let (num_txns_accepted, submit_duration_ms) = self.submit_txns(txn_reqs);
+        let rate = if let Some(p) = pattern {
```

Comment:
You can just `unwrap_or` and remove the whole if. For this you don't need to make it `mut`